lint-config.js

```diff
@@ -44,6 +45,7 @@ module.exports = {
 
   "rules": {
     "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn",
```

Comment:
Would honestly prefer we error on these but I don't want to break builds for people. Although now would be the time to do so if we're bumping up a major version

Reply:
i would still worry that this would become a blocker for people trying to upgrade 

maybe after the react 18 upgrade we can prioritize this 👍

Reply:
kk as long as there's at least a warning by default then I'm good with that

Reply:
I agree, I don't think we're ready quite yet, but i like the idea long term. I know some teams have made huge headway, but not everyone is there yet. its going to be much more important (also possibly easier) once we confirm all useEffects are safe when run multiple times